lib/nrf_modem_lib/shell/modem_trace_uart.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/uart.h>
+#include <modem/nrf_modem_lib_trace.h>
+
+#include "modem_trace_uart.h"
+
+#define UART_DEVICE_NODE DT_CHOSEN(nordic_modem_trace_uart)
+#define READ_BUF_SIZE CONFIG_NRF_MODEM_LIB_TRACE_SHELL_UART_BUF_SIZE
+static const struct device *const uart_dev = DEVICE_DT_GET(UART_DEVICE_NODE);
+
+void print_uart(const struct shell *sh, char *buf, int len)
+{
+	if (!device_is_ready(uart_dev)) {
+		shell_error(sh, "uart device not found/ready!");
+		return;
+	}
+	for (int i = 0; i < len; i++) {
+		uart_poll_out(uart_dev, buf[i]);
+	}
+}
+
+void modem_trace_uart_send(const struct shell *sh, size_t size)
+{
+	uint8_t read_buf[READ_BUF_SIZE];
```

Comment:
Since this isn't live processing of modem traces, but from flash, it should be fine to reduce the buffer size. Static memory is preferred to get build time errors instead of run-time. If the buffer is small, I guess the config isn't really needed anymore and can be removed.

Reply:
Depending on how big this buffer ends up being, it can be considered to move it off the stack while keeping it static. (Just add `static` to it).